src/components/LeaderboardAction.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="action-pill">
+    <BaseLink :to="{ name: path }" v-if="path">
+      <span class="mr-1">{{ text }} </span><BaseIcon :name="icon" />
+    </BaseLink>
+
+    <ShareButton :password="leaderboard.password" v-else />
```

Comment:
Just a style thing, but `v-if` and `v-else` should be the first thing right after `BaseLink`